### PR TITLE
[CALCITE-1966] Support for normal views to act as materialization table

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/prepare/Prepare.java [hunk 1]
```diff
@@ -143,10 +143,18 @@ protected RelRoot optimize(RelRoot root,
 
     final List<RelOptMaterialization> materializationList = new ArrayList<>();
     for (Materialization materialization : materializations) {
+      RelOptTable table = materialization.tableRel.getTable();
+      List<String> qualifiedTableName;
+      if (table == null) {
+        qualifiedTableName = materialization.materializedTable.path();
```
Comment (maryannxue) [inline core/src/main/java/org/apache/calcite/prepare/Prepare.java]: Just to confirm, can {{qualifiedTableName = materialization.materializedTable.path();}} work for both situations or this if-else block is inevitable?

File: core/src/main/java/org/apache/calcite/materialize/MaterializationService.java [hunk 2]
```diff
@@ -126,6 +126,10 @@ public MaterializationKey defineMaterialization(final CalciteSchema schema,
     if (tableEntry == null) {
       tableEntry = schema.getTableBySql(viewSql);
     }
+    if (tableEntry == null) {
```
Comment (maryannxue) [inline core/src/main/java/org/apache/calcite/materialize/MaterializationService.java]: Can we use the {{existing}} flag as we do for pre-populated materialization table, and move this if block into the {{if (existing)}} block?

File: core/src/main/java/org/apache/calcite/prepare/Prepare.java (hunk 1 above)
Comment (beikov) [reply core/src/main/java/org/apache/calcite/prepare/Prepare.java]: Seems it should work, I'll test that and push the change if it works.

File: core/src/main/java/org/apache/calcite/materialize/MaterializationService.java (hunk 2 above)
Comment (beikov) [reply core/src/main/java/org/apache/calcite/materialize/MaterializationService.java]: Sounds like a good idea.

**Pull request closed without merging**